servers/rendering/rasterizer_rd/shaders/canvas_uniforms_inc.glsl

```diff
@@ -110,37 +88,52 @@ struct Light {
 	mat2x4 texture_matrix; //light to texture coordinate matrix (transposed)
 	mat2x4 shadow_matrix; //light to shadow coordinate matrix (transposed)
 	vec4 color;
-	vec4 shadow_color;
-	vec2 position;
+
+	uint shadow_color; // packed
 	uint flags; //index to light texture
-	float height;
 	float shadow_pixel_size;
-	float pad0;
-	float pad1;
-	float pad2;
+	float height;
+
+	vec2 position;
+	float shadow_zfar_inv;
+	float shadow_y_ofs;
+
+	vec4 atlas_rect;
 };
 
-layout(set = 2, binding = 3, std140) uniform LightData {
+layout(set = 0, binding = 2, std140) uniform LightData {
 	Light data[MAX_LIGHTS];
 }
 light_array;
 
-layout(set = 2, binding = 4) uniform texture2D light_textures[MAX_LIGHT_TEXTURES];
-layout(set = 2, binding = 5) uniform texture2D shadow_textures[MAX_LIGHT_TEXTURES];
+layout(set = 0, binding = 3) uniform texture2D atlas_texture;
+layout(set = 0, binding = 4) uniform texture2D shadow_atlas_texture;
 
-layout(set = 2, binding = 6) uniform sampler shadow_sampler;
+layout(set = 0, binding = 5) uniform sampler shadow_sampler;
 
-#endif
+layout(set = 0, binding = 6) uniform texture2D screen_texture;
+
+layout(set = 0, binding = 7) uniform sampler material_samplers[12];
 
-layout(set = 2, binding = 7, std430) restrict readonly buffer GlobalVariableData {
+layout(set = 0, binding = 8, std430) restrict readonly buffer GlobalVariableData {
 	vec4 data[];
 }
 global_variables;
 
-/* SET3: Render Target Data */
+/* SET1: Is reserved for the material */
 
-#ifdef SCREEN_TEXTURE_USED
+//
 
-layout(set = 3, binding = 0) uniform texture2D screen_texture;
+/* SET2: Instancing and Skeleton */
 
-#endif
+layout(set = 2, binding = 0, std430) restrict readonly buffer Transforms {
+	vec4 data[];
```

Comment:
Looks like this doesn't include anything relating to skeleton

Reply:
This will contain either instance positions or skeleton transforms, like in the 3D version